tests/normalization_test.py

```diff
@@ -1,58 +1,102 @@
+
+    R = _SANS2D_PIXEL_RADIUS.value
+    L = _SANS2D_PIXEL_LENGTH.value
+    pixel_shape = {
+        'vertices': sc.vectors(
+            dims=['vertex'],
+            values=[
+                [0, 0, 0],
+                [R, 0, 0],
+                [0, L, 0],
+            ],
+            unit='m',
+        )
+    }
+    # Rotate counter-clockwise
+    transform = sc.spatial.rotation(value=[0, 0, 1 / 2**0.5, 1 / 2**0.5])
 
-def test_solid_angle():
-    l2 = np.arange(1.0, 11.0)
     da = sc.DataArray(
-        data=sc.array(dims=['x'], values=l2, unit='counts'),
-        coords={'L2': sc.array(dims=['x'], values=l2, unit='m')},
+        coords={'position': pixel_positions},
+        # Some dummy counts
+        data=sc.array(
+            dims=['spectrum'],
+            values=np.random.randint(0, 100, *pixel_positions.shape),
+            unit='counts',
+        ),
     )
-    da.coords['pixel_width'] = sc.scalar(2.0)
-    da.coords['pixel_height'] = sc.scalar(3.0)
+    return da, pixel_shape, transform
 
-    solid_angle = normalization.solid_angle_rectangular_approximation(da)
 
-    assert sc.isclose(solid_angle[0].data, solid_angle[1].data * 4).value
-    assert sc.isclose(solid_angle[0].data, solid_angle[-1].data * 100).value
-    da.coords['pixel_width'] *= 2
-    assert sc.allclose(
-        solid_angle.data * 2,
-        normalization.solid_angle_rectangular_approximation(da).data,
+def _mantid_sans2d_solid_angle_data():
+    simpleapi = pytest.importorskip("mantid.simpleapi")
+    scippneutron = pytest.importorskip("scippneutron")
+
+    ws = simpleapi.Load('SANS2D00063091.nxs')
+    radius = _SANS2D_PIXEL_RADIUS
+    length = _SANS2D_PIXEL_LENGTH
+
+    simpleapi.SetInstrumentParameter(
+        ws,
+        ParameterName='x-pixel-size',
+        ParameterType='Number',
+        Value=str((2 * radius).to(unit='mm').value),
+    )
+    simpleapi.SetInstrumentParameter(
+        ws,
+        ParameterName='y-pixel-size',
+        ParameterType='Number',
+        Value=str(length.to(unit='mm').value),
+    )
+    outWs = simpleapi.SolidAngle(ws, method='HorizontalTube')
+    da = scippneutron.from_mantid(outWs)['data']['spectrum', :120000:100]
+    # Recreate reference file:
+    # sc.io.hdf5.save_hdf5(da, _SANS2D_SOLID_ANGLE_REFERENCE_FILE_NAME)
+    return da
+
+
+@pytest.mark.skipif(
+    'SANS2D00063091.nxs' not in os.listdir(), reason='Needs data to recreate reference'
+)
+@pytest.mark.filterwarnings("ignore:.*")
+def test_solid_angle_compare_to_mantid():
+    da = _mantid_sans2d_solid_angle_data()
+    solid_angle = normalization.solid_angle(
+        *_sans2d_setup_requirements_for_solid_angle(da)
     )
-    da.coords['pixel_width'] *= 0.5
-    da.coords['pixel_height'] *= 3
     assert sc.allclose(
-        solid_angle.data * 3,
-        normalization.solid_angle_rectangular_approximation(da).data,
+        da.data['tof', 0], solid_angle, atol=1e-10 * sc.Unit('dimensionless')
     )
```

Comment:
So how large are the absolute and relative differences? How large are the values?

Reply:
Can you thus make `atol` and `rtol` as low as possible? This would make it clear that the results are really "the same".

Reply:
Why? Is a relative error of 1e-4 not sufficiently small? If we make the tolerances too low the test might break from a small change that doesn't meaningfully influence the result. The default `rtol` in allclose is 1e-5. So this is already about the same accuracy we have in every other place where `allclose` is used.